org.lflang/src/org/lflang/generator/GeneratorBase.xtend

```diff
@@ -1759,19 +1748,24 @@ abstract class GeneratorBase extends AbstractLFValidator implements TargetTypes
                         }
                     }
 
-                    FedASTUtils.makeCommunication(
-                        source,
-                        destination,
-                        connection,
-                        leftFederate,
-                        federate.bankIndex,
-                        source.index,
-                        rightFederate,
-                        destination.parent.bankIndex,
-                        destination.index,
-                        this,
-                        targetConfig.coordination
-                    )
+                    // Make one communication for each channel.
+                    // FIXME: There is an opportunity for optimization here by aggregating channels.
```

Comment:
Should we create an issue for this? I can take this up as an action item.

Reply:
It would be great if you would take this as an action item, but I suggest waiting until after scalability-banks is merged.